Mathlib/Algebra/Lie/LieTheorem.lean

```diff
@@ -0,0 +1,558 @@
+/-
+Copyright (c) 2024 Lucas Whitfield. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Lucas Whitfield
+-/
+import Mathlib.Algebra.Lie.Solvable
+import Mathlib.Algebra.Lie.Weights.Linear
+
+/-!
+# Lie's theorem for Solvable Lie algebras.
+-/
+set_option autoImplicit false
+
+open LieAlgebra
+
+
+-- let k be a field of characteristic zero
+variable {k : Type*} [Field k] [CharZero k] [IsAlgClosed k]
+-- Let L be a Lie algebra over k
+variable {L : Type*} [LieRing L] [LieAlgebra k L]
+-- and let V be a finite-dimensional k-representation of L
+variable {V : Type*} [AddCommGroup V] [Module k V] [FiniteDimensional k V]
+  [LieRingModule L V] [LieModule k L V] [Nontrivial V]
+
+open FiniteDimensional
+
+/- We define the Submodules generated by repeatedly applying a linear map `f: V →ₗ[F] V`
+to a vector `v`-/
+section
+
+variable {F V : Type*} [Field F] [AddCommGroup V] [Module F V] [FiniteDimensional F V]
+variable (v : V) (f : Module.End F V)
+-- the Submodule generated by `v`, `f v`, `f f v`, ... , `f^[n-1] v`
+abbrev U (n : ℕ) : Submodule F V := Submodule.span F {v' : V | ∃ a < n, v' = (f^a) v}
```

Comment:
This name will be hard to understand from the outside: can you give this a more descriptive name?
(You can use `local notation` to still name it `U` in this file --- but it will have a clearer name to the outside.)

Reply:
And also, this is currently in the root namespace, i.e. anyone who imports this and want to have `(U : Set X)` as an open space will get an error. Can you wrap the whole file in `namespace LieModule`, so that the theorems will be called `LieModule.*` outside this file.